BobClient/Qoollo.BobClient/BobNodeClient.cs

```diff
@@ -518,7 +519,157 @@ public Task<byte[]> GetAsync(ulong key)
         }
 
+        /// <summary>
+        /// Check data in Bob
+        /// </summary>
+        /// <param name="keys">Keys array</param>
+        /// <param name="fullGet">Try read data from sup nodes</param>
+        /// <param name="token">Cancellation token</param>
+        /// /// <returns>Operation result</returns>
+        /// <exception cref="ObjectDisposedException">Client was closed</exception>
+        /// <exception cref="TimeoutException">Timeout reached</exception>
+        /// <exception cref="OperationCanceledException">Operation was cancelled</exception>
+        /// <exception cref="BobOperationException">Other operation errors</exception>
+        public bool[] Exist(ulong[] keys, bool fullGet, CancellationToken token)
+        {
+            if (_isDisposed)
+                throw new ObjectDisposedException(GetType().Name);
+
+            var request = new BobStorage.ExistRequest(keys, fullGet);
+
+            try
+            {
+                var answer = _rpcClient.Exist(request, cancellationToken: token, deadline: GetDeadline(_operationTimeout));
+                return answer.Exist.ToArray();
+            }
+            catch (Grpc.Core.RpcException e)
+            {
+                if (IsOperationCancelledError(e, token))
+                    throw new OperationCanceledException(token);
+                if (IsOperationTimeoutError(e))
+                    throw new TimeoutException($"Exists operation timeout reached (node: {_nodeAddress}, speciefied timeout: {_operationTimeout})", e);
+                
+                throw new BobOperationException($"Exists operation failed.", e);
+            }
+        }
+
+        /// <summary>
+        /// Check data in Bob
+        /// </summary>
+        /// <param name="keys">Keys array</param>
+        /// <param name="token">Cancellation token</param>
+        /// <returns>Operation result</returns>
+        /// <exception cref="ObjectDisposedException">Client was closed</exception>
+        /// <exception cref="TimeoutException">Timeout reached</exception>
+        /// <exception cref="BobOperationException">Other operation errors</exception>
+        /// <exception cref="OperationCanceledException">Operation was cancelled</exception>
+        public bool[] Exist(ulong[] keys, CancellationToken token)
+        {
+            return Exist(keys, false, token);
+        }
 
+        /// <summary>
+        /// Check data in Bob
+        /// </summary>
+        /// <param name="keys">Keys array</param>
+        /// <param name="fullGet">Try read data from sup nodes</param>
+        /// <returns>Operation result</returns>
+        /// <exception cref="ObjectDisposedException">Client was closed</exception>
+        /// <exception cref="TimeoutException">Timeout reached</exception>
+        /// <exception cref="BobOperationException">Other operation errors</exception>
+        public bool[] Exist(ulong[] keys, bool fullGet)

```

Comment:
can we use fullGet=false and use single method?

Reply:
In my opinion - yes, but the old API was created with multiple methods, and my task was to add Exists like Get, so I did it like get methods are implemented. @ikopylov what do you think about that?

Reply:
It is not recommended to use default values in public API, because they do not supported by all .NET languages. That is why an overload methods used here.